packages/cozy-sharing/src/components/Recipient.jsx

```diff
@@ -75,142 +77,169 @@ export const UserAvatar = ({ url, size, ...rest }) => (
+              placement="bottom-end"
+              anchorElRef={buttonRef}
+              preventOverflow
             >
-              <div className={styles['action-unshare']}>
-                {isOwner
-                  ? t(`${documentType}.share.revoke.title`)
-                  : t(`${documentType}.share.revokeSelf.title`)}
-              </div>
-              <p className={styles['action-unshare-desc']}>
-                {isOwner
-                  ? t(`${documentType}.share.revoke.desc`)
-                  : t(`${documentType}.share.revokeSelf.desc`)}
-              </p>
-            </MenuItem>
-          </MenuAwareMobile>
-        )}
-      </div>
-    )
-  }
+              <ActionMenuItem
+                left={
+                  <Icon
+                    icon={permissionIconName}
+                    color="var(--primaryTextColor)"
+                  />
+                }
+              >
+                {t(`Share.type.${type}`)}
+              </ActionMenuItem>
+
+              <hr />
+              <ActionMenuItem
+                onClick={onRevokeClick}
+                left={<Icon icon="trash" color="var(--pomegranate)" />}
+              >
+                <Text className="u-pomegranate">
+                  {isOwner
+                    ? t(`${documentType}.share.revoke.title`)
+                    : t(`${documentType}.share.revokeSelf.title`)}
+                </Text>
+                <Caption>
+                  {isOwner
+                    ? t(`${documentType}.share.revoke.desc`)
+                    : t(`${documentType}.share.revokeSelf.desc`)}
+                </Caption>
+              </ActionMenuItem>
+            </ActionMenu>
+          )}
+        </>
+      )}
+    </div>
+  )
 }
 
-const StatusWithBreakpoints = withBreakpoints()(translate()(withClient(Status)))
+const Status = ({ status, isMe, instance }) => {
+  const { t } = useI18n()
+
+  const isError =
+    !isMe && ['error', 'unregistered', 'mail-not-sent'].includes(status)
+  const isReady = isMe || status === 'ready'
 
-const Recipient = (props, { client, t }) => {
+  let text, icon
+  if (isReady) {
+    text = instance
+    icon = 'to-the-cloud'
+  } else if (isError) {
+    text = t('Share.status.error')
+    icon = 'warning'
+  } else {
+    text = ['pending', 'seen'].includes(status)
+      ? t(`Share.status.${status}`)
+      : t('Share.status.pending')
+
+    icon = status === 'seen' ? 'eye' : 'paperplane'
+  }
+
+  return (
+    <Media>
+      <Img className={styles['recipient-status-icon']}>
+        <Icon icon={icon} size={10} />
+      </Img>
+      <Bd>
+        <Caption className={classNames({ 'u-pomegranate': isError })}>
+          {text}
+        </Caption>
+      </Bd>
+    </Media>
+  )
+}
+
+const Recipient = props => {
+  const { t } = useI18n()
+  const client = useClient()
   const { instance, isOwner, status, ...rest } = props
   const isMe =
     (isOwner && status === 'owner') || instance === client.options.uri
```

Comment:
Shouldn't we create an isMe() method since we also have it here : https://github.com/cozy/cozy-libs/pull/1057/files#diff-e3b0d831f7a76b61180beb6964f2ee6aR97 ?

Reply:
I'm reworking this, they don't really test the same thing.